File: base/src/main/java/com/windscribe/vpn/api/response/ClaimAccountResponse.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.vpn.api.response;

import androidx.annotation.NonNull;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;


@SuppressWarnings("unused")
public class ClaimAccountResponse {

    @SerializedName("password_updated")
    @Expose
    private int passwordUpdated;

    @SerializedName("success")
    @Expose
    private int success;

    @SerializedName("username_updated")
    @Expose
    private int usernameUpdated;

    public int getPasswordUpdated() {
        return passwordUpdated;
    }

    public int getUsernameUpdated() {
        return usernameUpdated;
    }

    public boolean isSuccessful() {
        return success == 1;
    }

    @NonNull
    @Override
    public String toString() {
        return "ClaimAccountResponse{" +
                "passwordUpdated=" + passwordUpdated +
                ", usernameUpdated=" + usernameUpdated +
                ", success=" + success +
                '}';
    }
}

File: base/src/main/java/com/windscribe/vpn/api/response/ClaimAccountResponse.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.vpn.api.response

import androidx.annotation.Keep
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

@Keep
class ClaimAccountResponse {

    @SerializedName("password_updated")
    @Expose
    val passwordUpdated = 0

    @SerializedName("success")
    @Expose
    private val success = 0

    @SerializedName("username_updated")
    @Expose
    val usernameUpdated = 0

    val isSuccessful: Boolean
        get() = success == 1

    override fun toString(): String {
        return "ClaimAccountResponse{" +
                "passwordUpdated=" + passwordUpdated +
                ", usernameUpdated=" + usernameUpdated +
                ", success=" + success +
                '}'
    }
}
